src/io/srt_io.ml

```diff
@@ -441,13 +441,12 @@ object (self)
           Srt.send socket data
       in
       Tutils.mutexify output_mutex (fun () ->
-        Buffer.blit buffer 0 tmp 0 payload_size;
-        Utils.buffer_drop buffer payload_size) ();
+          Strings.blit !buffer 0 tmp 0 payload_size;
+          buffer := Strings.drop !buffer payload_size) ();
       let rec f = function
         | pos when pos < payload_size ->
-          let ret =
-            send (Bytes.sub tmp pos (payload_size-pos))
-          in
+          (* TODO: it would be better if Srt.send / Srt.sendmsg would take an offset... *)
```

Comment:
@toots don't you think?

Reply:
The C API does a copy of the string to run without the global lock. I think that I have tried registering the string as global root but it didn't work. I'll give it another try,